### PR TITLE
feat(websdk): programmatic transaction system (demos.run)

### PR DESCRIPTION
**Pull request by tcsenpai**
## Summary

Adds a **programmatic transaction system** that collapses the classic
`build → sign → confirm → broadcast` flow into a single call per transaction
type, exposed as an additive `demos.run.*` facade. A single shared runner is
the only place that calls `confirm`/`broadcast`, normalising the three
historical builder shapes (returns-signed-tx, returns-validityData,
server-internal lifecycle) into one confirmation policy and one result type.

```ts
// before
const tx  = await demos.pay(to, amount)
const vd  = await demos.confirm(tx)
await demos.broadcast(vd)

// after — auto-broadcast within a configurable fee ceiling (default 5 DEM)
await demos.run.pay(to, amount)
```

The classic `demos.pay` / `confirm` / `broadcast` methods are **unchanged** —
zero breaking change.

## What's included

- **Shared runner** (`runProgrammaticTx`): `confirm → fee-cap / callback → broadcast`, uniform `ProgrammaticTxResult`.
- **Fee ceiling** (`maxFee`, default 5 DEM): auto-broadcast within the cap, else `FeeCapExceededError` (tx signed + confirmed but not broadcast). `confirm: "manual"` returns the validityData unbroadcast (classic flow); `confirm: (info) => boolean` for custom logic; `wait: true` for on-chain inclusion.
- **`Demos.start(rpc, wallet?)`**: one-call bootstrap over `connect` + `connectWallet`.
- **`broadcastAndWait`** default timeout 30s → 60s.
- **Namespaces**: `pay`/`transfer`, `attest.*` (github/twitter/discord/telegram/domain/removeWeb2 + web3-xm/pqc/ud/nomis/humanpassport/ethos/tlsn + dahr), `tokens.*`, `storage.*`, `escrow.*`, `validator.*`, `governance.*`, `xm.submit`, `bridge.submit`, `demoswork.submit`, `ipfs.*`, `d402.pay`, `contracts.*`.
- Docs: `documentation/programmatic-transactions.md`.

## Testing

- **17 unit tests** for the runner (fee wire-shape normalisation, fee cap, manual/callback/wait modes, source resolution, token-shape signing, `maxFee` edge values).
- **Live-verified against a node** (`node2.demos.sh`): `Demos.start`, `pay` (auto + manual), `storage.store`, `storage.program`, the fee cap firing at a real 2 DEM fee, and the lazy `run` getter. A live-testing pass surfaced a real bug (fee was read from `gas_operation.fees`, but the node carries it on `transaction.content.transaction_fee` when `gas_operation` is null) — fixed with a fallback + regression test.
- `bun run build` (full `tsc`) green on every commit.

## Pre-PR review — fixes applied

A multi-angle review ran before this PR. Fixes committed:
- **Import-cycle safety**: `demos.run` is now a lazy, memoised getter, not a constructor field — eagerly constructing the namespaces at the import-time `Demos.instance` singleton could crash on the websdk↔abstraction import cycle depending on entry-point order.
- **Token signing**: the runner now signs a tx unless its `signature.data` is actually populated (`DemosTokens` returns unsigned txs with a truthy-but-empty placeholder signature, which the old truthiness gate misread as signed).
- **Fee-cap edge**: only `null`/`Infinity` disable the cap; `NaN` now throws instead of silently removing the ceiling.
- **`isValidityData`** treats `response.data === null` as not-confirmed.

## Known caveats

- **`ipfs.*`, `d402.pay`, `contracts.*` and `tokens.*` are not enabled on production nodes yet** — wired and type-checked, but not exercisable end-to-end until the network turns them on.
- **`contracts.*`** is RPC-native and **not fee-capped** (no confirm/broadcast stage); it returns the contract's own result types, not `ProgrammaticTxResult`, and lives under `demos.run.contracts` purely for surface uniformity.
- **`xm.submit`** takes an already-assembled `XMScript` (per-chain payload construction stays with the caller); wired + type-checked but not live-verified.
- **Pre-existing (not changed here):** `DemosTokens` builders set `content.nonce` to the raw current nonce (not `nonce + 1` like every other builder), and the same `nonce` feeds `deriveTokenAddress`. This looks off-by-one against the sequential-nonce convention, but fixing it naively would desync the previewed token address from the on-chain one — left untouched pending verification against a token-enabled node.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01WGWnYw7UEXhvYawNm5zxWa)_

### CODE REVIEW

File: src/websdk/DemosTransactions.ts [hunk 1]
```diff
@@ -381,7 +381,7 @@ export const DemosTransactions = {
         broadcast: RPCResponse
         status: { state: "included" | "failed"; blockNumber?: number }
     }> {
-        const timeout = opts?.timeoutMs ?? 30_000
+        const timeout = opts?.timeoutMs ?? 60_000
```
[the remark this answers is not in the document]
Comment (tcsenpai) [reply src/websdk/DemosTransactions.ts:384]: Intentional and requested by the maintainer — the 30s→60s default is a deliberate change to absorb testnet inclusion latency (a 30s `broadcastAndWait` was timing out on txs that did land). Keeping it. Callers who need the old deadline can pass `{ timeoutMs: 30_000 }` explicitly; polling is `Date.now()`-based, so the only effect is a longer max wait, and no test asserts a ~30s deadline. Point taken on the "zero breaking change" wording — that claim is scoped to the classic `pay`/`confirm`/`broadcast` methods; the timeout bump is listed separately in the PR description as its own behavioral change.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by tcsenpai** [comment]
@greptile review

File: src/websdk/demosclass.ts [hunk 2]
```diff
@@ -40,6 +40,10 @@ import type { IBufferized } from "./types/IBuffer"
 import { IKeyPair } from "./types/KeyPair"
 import { _required as required } from "./utils/required"
 import { web2Calls } from "./Web2Calls"
+import {
+    createProgrammaticTx,
+    type ProgrammaticTx,
+} from "./programmatic"
```
[the remark this answers is not in the document]
Comment (tcsenpai) [reply src/websdk/demosclass.ts:46]: Correct that the lazy getter defers *construction*, not module *loading* — a static ESM import can't be deferred. Deciding not to refactor to async/dynamic-import loading, because:

- The cycle bug (#1) is resolved: the crash vector was eager *construction* in the constructor, which the lazy getter removes.
- The residual cycle is benign — no programmatic module reads a cross-cycle binding at module-eval time (only inside functions), so there are no init-order/undefined-export hazards.
- It's been narrowed: `attest.ts` now imports `Identities` directly from `@/abstraction/Identities` instead of the barrel, keeping the coin-finder/provider surface out of the eager load.

Fully deferring the module load in ESM would mean per-namespace `await import(...)`, which would make the synchronous `demos.run.*` surface async — a real API regression for a benign, non-blocking cycle. Updated the getter's JSDoc to state this accurately rather than overclaim that laziness sidesteps the cycle.

---
_Generated by [Claude Code](https://claude.ai/code)_
